packages/functions/src/dataTriggers.ts

```diff
@@ -269,7 +269,84 @@ export const createAttendanceForBooking = functions
 
     await attendanceRef.set(updatedEntry, { merge: true });
   });
+// SlotAttendnace = {
+//   /**
+//    * ISO date of the slot for easier querying
+//    */
+//   date: string;
+//   /**
+//    * Attendances for slot keyed by customer
+//    */
+//   attendances: {
+//     [customerId: string]: CustomerAttendance;
+//   };
+// };
+
+// CustomerAttendance {
+//   bookedInterval: string | null;
+//   attendedInterval: string | null;
+// }
+
+// entry
+// [slotId: string]: CustomerBookingEntry;
+// {
+//   /**
+//    * ISO date of the booked slot (used for easier querying from store)
+//    */
+//   date: string;
+//   /**
+//    * Booked interval for particular slot
+//    */
+//   interval: string;
+
+// }
+export const createAttendedSlotOnAttendance = functions
+  .region(__functionsZone__)
+  .firestore.document(
+    `${Collection.Organizations}/{organization}/${OrgSubCollection.Attendance}/{bookingId}/`
+  )
+  .onWrite(async (change, context) => {
+    const { organization, bookingId } = context.params as Record<
+      string,
+      string
+    >;
+
+    const db = admin.firestore();
 
+    // const isCreate = !change.before.exists;
+    const isUpdate = change.after.exists;
+    // const isDelete = !change.after.exists;
+
+    if (isUpdate) {
+      const attendanceData = change.after.data() as SlotAttendnace;
+
+      Object.keys(attendanceData.attendances).forEach(async (customerId) => {
+        const attendances = attendanceData.attendances[customerId];
```

Comment:
It is better to replace `.forEach` with `.map` and return 0 or null, or whatever from each map. Then to `Promise.all` on the result, e.g.:
```typescript
const updates = someArray.map(async () => {
  await someAsyncTask()
  return null
}

await Promise.all(updates)
```

Reply:
This returns an array of promises which all resolve to `null`, but `Promise.all` can be utilised to await all of the async tasks

Reply:
Or better yet:
```typescript
const updates = someValues.map(() => new Promise<void>(async (resolve, reject) => {
  await someAsyncTask()
  resolve()
}))
```

This way you're explicit about returning a promise from each map function and are not relying on syntactic sugar which might, or might not behave as you assume, so I prefer that approach.

_notice no `async` in the map function and `async` being introduced only in the promise callback_